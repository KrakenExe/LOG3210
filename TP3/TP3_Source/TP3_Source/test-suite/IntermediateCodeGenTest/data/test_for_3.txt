Declaration
num a;
num i;
num c;
bool d;
EndDeclaration

for(c = 0; a > 1; i = i + 1) {
    a = 1 + 1;
    if(false) {
      a = 42;
    }
    else {
      a = 3210;
      while(!d) {
        a = a + 42;
        c = 1 / 3 + 5;
      }
    }

    for(c = 0; a > 1; i = i + 1) {
        a = -3210;
    }
}