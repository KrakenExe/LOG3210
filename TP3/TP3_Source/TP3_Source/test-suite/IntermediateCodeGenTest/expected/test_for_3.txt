c = 0
_L1
if a > 1 goto _L2
goto _L0
_L2
_t0 = 1 + 1
a = _t0
_L3
goto _L6
_L5
a = 42
goto _L4
_L6
a = 3210
_L7
_L8
if d == 1 goto _L4
goto _L9
_L9
_t1 = a + 42
a = _t1
_L10
_t3 = 1 / 3
_t2 = _t3 + 5
c = _t2
goto _L8
_L4
c = 0
_L11
if a > 1 goto _L12
goto _L1
_L12
_t4 = - 3210
a = _t4
_t5 = i + 1
i = _t5
goto _L11
_t6 = i + 1
i = _t6
goto _L1
_L0
